src/XMakeTasks/Microsoft.CSharp.Mono.targets

```diff
@@ -0,0 +1,36 @@
+﻿<!--
+***********************************************************************************************
+Microsoft.CSharp.Mono.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+This file defines the steps in the standard build process specific for C# .NET projects.
+For example, it contains the step that actually calls the C# compiler.  The remainder
+of the build process is defined in Microsoft.Common.targets, which is imported by 
+this file.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <!-- Using _CscToolExeTmp instead of CscToolExe directly here, as we don't want to override it
+             for !mono case -->
+        <_CscToolExeTmp Condition="'$(CscToolExe)' != ''">$(CscToolExe)</_CscToolExeTmp>
+        <_CscToolExeTmp Condition="'$(CscToolExeTmp)' == '' and '$(MSBuildRuntimeType)' == 'Mono'">mcs.exe</_CscToolExeTmp>
+
+        <CscDebugFileExt Condition="'$(CscDebugFileExt)' == '' and ('$(_CscToolExeTmp)' == 'mcs' or '$(_CscToolExeTmp)' == 'mcs.exe')">.mdb</CscDebugFileExt>
+        <_DebugFileExt Condition="'$(CscDebugFileExt)' != ''">$(CscDebugFileExt)</_DebugFileExt>
+
+        <!-- Select a default C# compiler here, if unspecified, instead of in the Csc task, so that we can also
+             decide on the debug extension to be used. Don't do this for the roslyn/csc case as it depends on the
+             default being "unset".
+             -->
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' == 'Mono'">mcs.exe</CscToolExe>
```

Comment:
Yep, it would fail in that case! But I consider that a user error, since they are explicitly opting to use `.mdb` as the debug file extension! This is meant for the case where - defaults work, and if the user wants to use some custom compiler then they can and be able to override the debug file extension too.

The compiler selection is required here for being able to select a debug file extension. In the default (non-mono) case, it will work as is - csc.exe/.pdb . In case of mono, if you are not using Rosyln, then it will work as mcs.exe/.mdb . If users want to use Rosyln, then they override the two properties and can use it.

From the commit message:

> - Providing default values for the `*DebugFileExt` requires knowing which
  compiler is being used, but if `$(CscToolExe)` is not set (by the user
  or in the project files), then we need some defaults.

>  So, we try to set reasonable *defaults* for `$(CscToolExe)` based on
  the runtime host. This still allows the user to override that,
  whether in the project/targets file or via the command line. This is
  done in new `Microsoft.{CSharp,VisualBasic}.Mono.targets` file and the
  override is set *only* for the mono case.